lib/ansible/module_utils/facts.py

```diff
@@ -915,6 +927,104 @@ def get_memory_facts(self):
         self.facts['swap_allocated_mb'] = allocated / 1024
         self.facts['swap_reserved_mb'] = reserved / 1024
 
+    def get_dmi_facts(self):
+        smbios_bin = module.get_bin_path('smbios')
+
+        SMB_TYPE_BIOS = dict(
+            bios_date='Release Date',
+            bios_version='Version String',
+        )
+        SMB_TYPE_SYSTEM = dict(
+            product_name='Product',
+            product_serial='Serial Number',
+            product_uuid='UUID',
+            product_version='Version',
+            system_vendor='Manufacturer'
+        )
+
+        SMB_TYPE_CHASSIS = dict(
+            form_factor='Chassis Type',
+        )
+
+        for (k, v) in SMB_TYPE_BIOS.items():
+            if smbios_bin is not None:
+                (rc, out, err) = module.run_command('%s -t SMB_TYPE_BIOS' % (smbios_bin))
+                if rc == 0:
+                    smbios_dict = dict((line.split(':')[0].strip(), line.split(':')[1].strip()) for line in out.splitlines() if ':' in line )
+                    thisvalue = smbios_dict[v]
+                    try:
+                        json.dumps(thisvalue)
+                    except UnicodeDecodeError:
+                        thisvalue = "NA"
```

Comment:
I see that other get_dmi_decode() methods are using this pattern as well but it seems very inefficient as well.  json.dumps() is not cheap and here we're just throwing away the value.  Perhaps it would be better to do something like:
``` python
if isinstance(str, thisvalue):
    thisvalue = unicode(thisvalue, encoding='utf-8', errors='replace').encode('utf-8')
```